ni_measurementlink_service/_internal/parameter/serializer.py

```diff
@@ -75,15 +76,17 @@ def serialize_parameters(
             parameter_metadata.type,
             parameter_metadata.repeated,
         )
-        # Convert enum parameters to their underlying value.
+        # Convert enum parameters to their underlying value if necessary.
         if (
             parameter_metadata.annotations.get("ni/type_specialization")
             == TypeSpecialization.Enum.value
         ):
             if parameter_metadata.repeated:
-                parameter = [x.value for x in parameter]
+                if len(parameter) > 0 and isinstance(parameter[0], Enum):
+                    parameter = [x.value for x in parameter]
             else:
-                parameter = parameter.value
+                if isinstance(parameter, Enum):
+                    parameter = parameter.value
```

Comment:
It would be more readable to call a `_get_enum_value` function that does the `isinstance` check because then you don't have to think about what it means to skip assigning to `parameter`.

Reply:
Added `_get_enum_value` helper as suggested.